doc/command_guidelines.md

```diff
@@ -0,0 +1,63 @@
+# Command Guidelines
+
+This document describes the command guidelines for 'az' and applies to both CLI command modules and extensions.
+
+Guidelines marked (*) only apply to command modules, not extensions.
+
+If in doubt, ask!
+
+## General Patterns
+
+- Be consistent with POSIX tools (support piping, work with grep, awk, jq, etc.)
+- Support tab completion for parameter names and values (e.g. resource names)
+- Commands must follow a "[noun] [noun] [verb]" pattern
```

Comment:
Consider moving this down into the "Command Naming and Behavior Guidance" section.

Reply:
done